src/py42/services/alertrules.py

```diff
@@ -0,0 +1,136 @@
+            ],
+        }
+        uri = u"{}{}".format(self._api_prefix, u"add-users")
+        return self._connection.post(uri, json=data)
+
+    def remove_user(self, rule_id, user_id):
+        user_ids = [user_id]
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {u"tenantId": tenant_id, u"ruleId": rule_id, u"userIdList": user_ids}
+        uri = u"{}{}".format(self._api_prefix, u"remove-users")
+        return self._connection.post(uri, json=data)
+
+    def remove_all_users(self, rule_id):
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {u"tenantId": tenant_id, u"ruleId": rule_id}
+        uri = u"{}{}".format(self._api_prefix, u"remove-all-users")
+        return self._connection.post(uri, json=data)
+
+
+class CloudShareService(BaseService):
+
+    _version = u"v1"
+    _resource = u"query-cloud-share-permissions-rule"
+    _api_prefix = u"/svc/api/{}/Rules/{}".format(_version, _resource)
+
+    def __init__(self, connection, tenant_id):
+        super(CloudShareService, self).__init__(connection)
+        self._tenant_id = tenant_id
+
+    def get(self, rule_id):
+        """Fetch cloud share alert rule by rule id.
+
+        Args:
+            rule_id (str): Observer rule Id of a rule to be fetched.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        data = {u"tenantId": self._tenant_id, u"ruleIds": [rule_id]}
+        return self._connection.post(self._api_prefix, json=data)
+
+
+class ExfiltrationService(BaseService):
+
+    _version = u"v1"
+    _resource = u"query-endpoint-exfiltration-rule"
+    _api_prefix = u"/svc/api/{}/Rules/{}".format(_version, _resource)
+
+    def __init__(self, connection, tenant_id):
+        super(ExfiltrationService, self).__init__(connection)
+        self._tenant_id = tenant_id
+
+    def get(self, rule_id):
+        """Fetch exfiltration alert rule by rule id.
+
+        Args:
+            rule_id (str): Observer rule Id of a rule to be fetched.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        data = {u"tenantId": self._tenant_id, u"ruleIds": [rule_id]}
+        return self._connection.post(self._api_prefix, json=data)
+
+
+class FileTypeMismatchService(BaseService):
+
+    _version = u"v1"
+    _resource = u"query-file-type-mismatch-rule"
+    _api_prefix = u"/svc/api/{}/Rules/{}".format(_version, _resource)
+
+    def __init__(self, connection, tenant_id):
+        super(FileTypeMismatchService, self).__init__(connection)
+        self._tenant_id = tenant_id
+
+    def get(self, rule_id):
+        """Fetch File type mismatch alert rules by rule id.
+
+        Args:
+            rule_id (str): Observer rule Id of a rule to be fetched.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        data = {"tenantId": self._tenant_id, "ruleIds": [rule_id]}
```

Comment:
Is this an example of where the strings may be concat'ed and is dangerous not to have `u`?

Reply:
for dict keys it would probably not be a concern unless we expected a use case where those keys would be concatenated with a unicode string. A better example would be some thing like an exception message that contained a file name -- there's a decent change the file name could have unicode characters, which would cause building the error message to fail when the file name is appended to the (non-unicode) rest of the message.